src/middlewared/middlewared/alembic/versions/21.MM/2021-08-20_17-20_serial_ports.py

```diff
@@ -0,0 +1,37 @@
+"""
+User serial port(s) name instead of I/O address
+
+Revision ID: 725b7264abe6
+Revises: 29abd3dce632
+Create Date: 2021-20-08 17:20:42.818433+00:00
+"""
+from alembic import op
+import sqlalchemy as sa
+
+from middlewared.utils import osc
+
+
+revision = '725b7264abe6'
+down_revision = '29abd3dce632'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    conn = op.get_bind()
+    io_choices = {e['start']: e['name'] for e in osc.system.serial_port_choices()}
+    sys_config = [dict(row) for row in conn.execute("SELECT * FROM system_advanced").fetchall()]
+    if not sys_config:
+        return
+
+    sys_config = sys_config[0]
+    if sys_config['adv_serialport'] not in io_choices and not sys_config['adv_serialport'].startswith('ttyS'):
+        new_val = 'ttyS0'
+    else:
+        new_val = io_choices[sys_config['adv_serialport']]
```

Comment:
It will crash if `(sys_config['adv_serialport'] not in io_choices) == True` and `not sys_config['adv_serialport'].startswith('ttyS') == False`